tests/translator/input/api_with_custom_base_path.yaml

```diff
@@ -32,3 +32,18 @@ Resources:
           HostedZoneId:
             Ref: HostedZoneId
           IpV6: true
+
+  MyApiWithoutStripHyphens:
+    Type: AWS::Serverless::Api
+    Properties:
+      OpenApiVersion: 3.0.1
+      StageName: Prod
+      Domain:
+        DomainName: MyEdgeDomainName
+        CertificateArn: MyEdgeDomainCert
+        EndpointConfiguration: EDGE
+        StripHyphens: false
```

Comment:
1. Is `StripHyphens` clear enough from context? An uglier but clearer alternative could be `BasePathStripHyphens`.
2. Is `StripHyphens` correct? I don't see code specifically stripping hyphens, but anything non-alphanumeric.

Reply:
Good call. `BasePathStripSymbols` is probably better in this context.

Reply:
Symbol is quite a fancy word, and makes me think of symbol tables in binaries. Plus any character can really be a "symbol".

Some bad alternatives:

- `BasePathSanitize`/`SanitizeBasePath`
- `BasePathAlphanumeric`